### PR TITLE
Clean up direct ->reply access

### PR DESCRIPTION
**Pull request by danquah**

### CODE REVIEW

File: modules/ding_provider/connie/connie.loan.inc [hunk 1]
```diff
@@ -45,3 +47,21 @@ function connie_loan_renew($account, $ids) {
...
+ *   Mutable ting object that represents the raw search result from the search
+ *   provider.
+ *
+ * @return object
+ *   Ting entity object, filled with data from provider service.
+ */
+function connie_loan_populate_pseudo_entity($ting_entity, TingObject $ting_object) {
+  $ting_object->setTitle(trim($ting_object->getTitle() . " Connie edition"));
+
+  return $ting_entity;
```
Comment (kasperg) [inline modules/ding_provider/connie/connie.loan.inc:66]: This modifies `$ting_object` but returns `$ting_entity`. It is not clear how/if the entity has been modified. Perhaps it is an oddity based on the class contracts where the ting entity cannot return a mutable object 🤔 .

File: modules/ding_provider/connie/connie.module [hunk 2]
```diff
@@ -141,8 +141,18 @@ function connie_availability_holdings($provider_ids) {
...
+          'reservable' => TRUE,
+          'total_count' => 10,
+          'available' => TRUE,
+          'ordered_count' => 2,
+        ],
+        'Gåserød bibliotek' => [
+          'reservable' => TRUE,
+          'total_count' => 10,
+          'available' => TRUE,
+          'ordered_count' => 2,
```
Comment (kasperg) [inline modules/ding_provider/connie/connie.module]: Having different values for `reservable`, `total_count` etc. for the different branch holdings would be useful.

File: modules/opensearch/src/OpenSearchTingObject.php [hunk 3]
```diff
@@ -721,7 +728,7 @@ private function processSeriesDescription($series) {
    *   relations has been loaded.
    */
   protected function getRelationsData() {
-    return isset($this->openSearchObject->relationsData) ? $this->openSearchObject->relationsData : [];
+    return isset($this->openSearchObject->relationsData) ? $this->openSearchObject->relationsData : NULL;
```
Comment (kasperg) [inline modules/opensearch/src/OpenSearchTingObject.php:731]: :+1:

File: modules/ting/ting.entities.inc [hunk 4]
```diff
@@ -24,17 +25,24 @@ class TingRelation extends DingEntityBase {
     }
   }
 
+  /**
+   * Ting Entity if the relation is a relation to a material.
+   *
+   * @return \TingEntity|null
```
Comment (kasperg) [inline modules/ting/ting.entities.inc]: Perhaps `NULL`  for consistency?

File: modules/ting/ting.entities.inc [hunk 5]
```diff
@@ -78,6 +86,26 @@ class TingEntity extends DingEntity {
    */
   public $reply;
```
Comment (kasperg) [inline modules/ting/ting.entities.inc:99]: Can we change the visbility of the property now? I do not think we want to leave it as public.

File: modules/ting/ting.entities.inc [hunk 6]
```diff
@@ -346,6 +374,16 @@ class TingCollection extends DingEntityCollection {
    */
   public $tingObjectCollection;
```
Comment (kasperg) [inline modules/ting/ting.entities.inc:387]: See not about public properties above.

File: modules/ting_fulltext/ting_fulltext.module [hunk 7]
```diff
@@ -7,6 +7,8 @@
  * This uses a docbook-subset for displaying content.
  */
 
+use OpenSearch\OpenSearchTingObject;
```
Comment (kasperg) [inline modules/ting_fulltext/ting_fulltext.module:10]: Perhaps this module should require `opensearch` if it does not already do so?

File: modules/ting_new_materials/ting_new_materials.module [hunk 8]
```diff
@@ -168,6 +168,8 @@ function ting_new_materials_do_search($query, $page = 0, $num_per_page = 10) {
   $search_result = array();
   try {
     // TODO: Consider porting this to the Search Abstraction Layer.
+    // When doing so, remember to clean out all use of ->reply eg in
+    // ting_new_materials_process_search_results.
```
Comment (kasperg) [inline modules/ting_new_materials/ting_new_materials.module:172]: 😞

File: modules/ting/ting.entities.inc (hunk 5 above)
Comment (kasperg) [reply modules/ting/ting.entities.inc:99]: As I continue reading I can tell that unported modules mean we cannot change this yet...

**Review by kasperg — APPROVED** [review]
:+1: Looks good to me.

I have some comments but they should not block merging.

File: modules/ding_provider/connie/connie.loan.inc (hunk 1 above)
Comment (danquah) [reply modules/ding_provider/connie/connie.loan.inc:66]: yeah, that's pretty much it. I could actually just mutate $ting_entity and return nothing, but as the ding_provider.module assigns the return-value to its copy of $ting_entity that would end up overwriting the entity.

The alternative would be to just forego with returning anything and relying on pass-by reference.

... I'll give it a quick try.

File: modules/ting/ting.entities.inc (hunk 5 above)
Comment (danquah) [reply modules/ting/ting.entities.inc:99]: :+1: to the "not yet"

**Pull request merged**